src/frontend/src/controllers/API/queries/flows/use-get-refresh-flows.ts

```diff
@@ -1,67 +1,92 @@
+import buildQueryStringUrl from "@/controllers/utils/create-query-param-string";
 import useAlertStore from "@/stores/alertStore";
 import useFlowsManagerStore from "@/stores/flowsManagerStore";
-import { useFolderStore } from "@/stores/foldersStore";
 import { useTypesStore } from "@/stores/typesStore";
 import { useMutationFunctionType } from "@/types/api";
-import { FlowType } from "@/types/flow";
+import { FlowType, PaginatedFlowsType } from "@/types/flow";
 import {
   extractFieldsFromComponenents,
   processFlows,
 } from "@/utils/reactflowUtils";
-import { UseMutationResult } from "@tanstack/react-query";
+import { UseMutationOptions } from "@tanstack/react-query";
 import { api } from "../../api";
 import { getURL } from "../../helpers/constants";
 import { UseRequestProcessor } from "../../services/request-processor";
 
+interface GetFlowsParams {
+  components_only?: boolean;
+  get_all?: boolean;
+  header_flows?: boolean;
+  folder_id?: string;
+  remove_example_flows?: boolean;
+  page?: number;
+  size?: number;
+}
+
+const addQueryParams = (url: string, params: GetFlowsParams): string => {
+  return buildQueryStringUrl(url, params);
+};
+
 export const useGetRefreshFlows: useMutationFunctionType<
   undefined,
-  undefined
-> = (options?) => {
+  GetFlowsParams
+> = (options) => {
   const { mutate } = UseRequestProcessor();
 
-  const setExamples = useFlowsManagerStore((state) => state.setExamples);
   const setFlows = useFlowsManagerStore((state) => state.setFlows);
   const setErrorData = useAlertStore((state) => state.setErrorData);
-  const starterProjectId = useFolderStore((state) => state.starterProjectId);
 
-  const getRefreshFlowsFn = async (): Promise<void> => {
+  const getRefreshFlowsFn = async (
+    params: GetFlowsParams,
+  ): Promise<FlowType[] | PaginatedFlowsType> => {
+    const url = addQueryParams(`${getURL("FLOWS")}/`, params);
+    const { data } = await api.get<FlowType[]>(url);
+    return data;
+  };
+
+  const mutationFn = async (
+    params?: GetFlowsParams,
+  ): Promise<FlowType[] | PaginatedFlowsType> => {
     try {
+      await getRefreshFlowsFn(params!).then(async (dbDataFlows) => {
+        const dbDataComponents = await getRefreshFlowsFn({
+          components_only: true,
+          get_all: true,
+        });
+
+        if (dbDataComponents) {
+          const { data } = processFlows(dbDataComponents as FlowType[]);
+          useTypesStore.setState((state) => ({
+            data: { ...state.data, ["saved_components"]: data },
+            ComponentFields: extractFieldsFromComponenents({
+              ...state.data,
+              ["saved_components"]: data,
+            }),
+          }));
+        }
+
+        if (dbDataFlows) {
+          const flows = Array.isArray(dbDataFlows)
+            ? dbDataFlows
+            : dbDataFlows.items;
+
+          setFlows(flows);
```

Comment:
I believe this needs to filter out template flows as even if a users has no flows creates a new flow from a template it will include a `(1)`

Reply:
@mfortman11 The name of the flows must be unique for each user and not by each folder... I think that's why you are seeing the` (1) ` included